enterprise/backend/src/metabase_enterprise/gsheets/settings.clj

```diff
@@ -1,34 +1,54 @@
 (ns metabase-enterprise.gsheets.settings
   (:require
+   [clojure.set :as set]
    [metabase-enterprise.gsheets.constants :as gsheets.constants]
    [metabase.models.setting :as setting :refer [defsetting]]
    [metabase.premium-features.core :as premium-features]
    [metabase.util :as u]
    [metabase.util.i18n :refer [deferred-tru]]
    [metabase.util.json :as json]
+   [metabase.util.malli :as mu]
    [metabase.util.malli.registry :as mr]
    [metabase.util.malli.schema :as ms]
    [toucan2.core :as t2]))
 
-(mr/def :gsheets/gsheets
-  [:multi {:dispatch :status}
-   ["not-connected" [:map]]
+(mr/def :gsheets/gsheets-response
+  [:or
+   [:map
+    [:error true?]
+    [:message ms/NonBlankString]]
+   [:multi {:dispatch :status}
+    ["not-connected" [:map]]
 
-   ["loading"
-    [:map
-     [:folder_url ms/NonBlankString]
-     ;; time in seconds from epoch:
-     [:folder-upload-time pos-int?]
-     [:gdrive/conn-id ms/UUIDString]
-     [:created-by-id pos-int?]]]
+    ["syncing"
+     [:map
+      [:url ms/NonBlankString]
+      ;; time in seconds from epoch:
+      [:created_at pos-int?]
+      ;; time in seconds from epoch:
+      [:sync_started_at pos-int?]
+      [:created_by_id pos-int?]]]
 
-   ["complete"
-    [:map
-     [:folder_url ms/NonBlankString]
-     ;; time in seconds from epoch:
-     [:folder-upload-time pos-int?]
-     [:gdrive/conn-id ms/UUIDString]
-     [:created-by-id pos-int?]]]])
+    ["active"
+     [:map
+      [:url ms/NonBlankString]
+      ;; time in seconds from epoch:
+      [:created_at pos-int?]
+      ;; time in seconds from epoch:
+      [:last_sync_at pos-int?]
+      ;; time in seconds from epoch:
+      [:next_sync_at pos-int?]
+      [:created_by_id pos-int?]]]]])
+
+(mr/def :gsheets/gsheets-setting
```

Comment:
I think it would read better to rename `:gsheets/gsheets-setting` to `:gsheets/setting`

Reply:
same with :gsheets/gsheets-response -> :gsheets/response